packages/foam-vscode/src/features/document-decorator.ts

```diff
@@ -14,34 +14,59 @@ const placeholderDecoration = vscode.window.createTextEditorDecorationType({
   cursor: 'pointer',
 });
 
+const tagDecoration = vscode.window.createTextEditorDecorationType({
+  rangeBehavior: vscode.DecorationRangeBehavior.ClosedClosed,
+  textDecoration: 'none',
+  color: { id: 'foam.tag' },
+  cursor: 'pointer',
+});
+
 const updateDecorations = (
   parser: ResourceParser,
   workspace: FoamWorkspace
 ) => (editor: vscode.TextEditor) => {
   if (!editor || editor.document.languageId !== 'markdown') {
     return;
   }
+
   const note = parser.parse(
     fromVsCodeUri(editor.document.uri),
     editor.document.getText()
   );
-  const placeholderRanges = [];
-  note.links.forEach(link => {
-    const linkUri = workspace.resolveLink(note, link);
-    if (linkUri.isPlaceholder()) {
-      placeholderRanges.push(
-        Range.create(
-          link.range.start.line,
-          link.range.start.character + (link.type === 'wikilink' ? 2 : 0),
-          link.range.end.line,
-          link.range.end.character - (link.type === 'wikilink' ? 2 : 0)
-        )
-      );
-    }
-  });
-  editor.setDecorations(placeholderDecoration, placeholderRanges);
+
+  editor.setDecorations(
+    placeholderDecoration,
+    getPlaceholderRanges(workspace, note)
+  );
+
+  editor.setDecorations(tagDecoration, getTagRanges(note.tags));
 };
 
+const getPlaceholderRanges = (
+  workspace: FoamWorkspace,
+  note: Resource
+): vscode.Range[] =>
+  note.links
+    .filter(link => workspace.resolveLink(note, link).isPlaceholder())
+    .map(link =>
+      Range.create(
+        link.range.start.line,
+        link.range.start.character + (link.type === 'wikilink' ? 2 : 0),
+        link.range.end.line,
+        link.range.end.character - (link.type === 'wikilink' ? 2 : 0)
+      )
+    ) as vscode.Range[];
+
+const getTagRanges = (tags: Tag[]): vscode.Range[] =>
+  tags.map(tag =>
+    Range.create(
```

Comment:
you should be able to use the `toVsCodeRange` function here

Reply:
Good call 👍 Done